library/CMakeLists.txt

```diff
@@ -377,6 +377,9 @@ if(WIN32)
     set_target_properties(dfhack-client PROPERTIES COMPILE_FLAGS "/FI\"Export.h\"" )
 else()
     set_target_properties(dfhack PROPERTIES COMPILE_FLAGS "-include Export.h" )
+    # required because of protobuf headers
+    target_compile_options(dfhack
+        PUBLIC -Wno-deprecated-declarations -Wno-restrict)
```

Comment:
Does setting these flags as PUBLIC on the protobuf targets not work? I thought something along those lines was supposed to work.

If this is required for DFHack, is it also required for plugins?

Reply:
I believe it's required here because while plugins include protobuf headers, they do not directly link against the protobuf libs, so they don't pick up the flags.

A better solution might be to not include protobuf headers from DFHack headers

Reply:
> If this is required for DFHack, is it also required for plugins?

It's actually not required for DFHack, since it picks up the flags from protobuf. I could have set the DFHack flags as INTERFACE instead of PUBLIC

Reply:
changed to INTERFACE

Reply:
> It's actually not required for DFHack

I'm confused, then, why set the flags on the `dfhack` target at all? Is the issue that plugins won't pick up on INTERFACE flags set on the `protobuf` target, but will pick up on flags set on `dfhack`?

I'm also not a huge fan of disabling warnings globally due to an issue in a third-party dependency. Since we have full control over our copies of the protobuf headers, would we be able to add `#pragma GCC diagnostic ignored` and `#pragma GCC diagnostic pop` around the relevant bits? Or maybe this would work, which we use in DF headers to silence most warnings - unsure if this covers the warnings in question or not:
```c
#ifdef __GNUC__
#pragma GCC system_header
#endif
```